src/main/java/io/logz/logback/LogzioSender.java

```diff
@@ -47,25 +49,38 @@
 
     public LogzioSender(String logzioToken, String logzioType, int drainTimeout, int fsPercentThreshold, String bufferDir,
                         String logzioUrl, int socketTimeout, int connectTimeout, boolean debug,
-                        LogzioLogbackAppender.StatusReporter reporter, ScheduledExecutorService tasksExecutor) throws IllegalArgumentException {
+                        LogzioLogbackAppender.StatusReporter reporter, ScheduledExecutorService tasksExecutor, boolean addHostname, String additionalFields) throws IllegalArgumentException {
+
+        this.logzioToken = logzioToken;
+        this.logzioType = logzioType;
+        this.drainTimeout = drainTimeout;
+        this.fsPercentThreshold = fsPercentThreshold;
+        this.logzioUrl = logzioUrl;
+        this.socketTimeout = socketTimeout;
+        this.connectTimeout = connectTimeout;
+        this.debug = debug;
+        this.reporter = reporter;
+
+        if (this.fsPercentThreshold == -1) {
+            dontCheckEnoughDiskSpace = true;
+        }
+
+        logsBuffer = new BigQueue(bufferDir, "logzio-logback-appender");
+        queueDirectory = new File(bufferDir);
+
+        if (additionalFields != null) {
+            this.additionalFields = Splitter.on(';').omitEmptyStrings().withKeyValueSeparator('=').split(additionalFields);
```

Comment:
Maybe catch format exceptions and throw a good exception with bad var name and suggested format?